### PR TITLE
Added sample for building Native Java Langchain4J applications with GraalVM

### PR DESCRIPTION
**Pull request by ddobrin**
Demo uses Function Calling from Vertex Gemini

### CODE REVIEW

File: native-java-gemini-function-calling-example/pom.xml [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+	<modelVersion>4.0.0</modelVersion>
+	<parent>
+		<groupId>org.springframework.boot</groupId>
+		<artifactId>spring-boot-starter-parent</artifactId>
+		<version>3.3.0</version>
+		<relativePath/>
+		<!-- lookup parent from repository -->
+	</parent>
+
+	<groupId>spring.ai.demo</groupId>
```
Comment (langchain4j) [inline native-java-gemini-function-calling-example/pom.xml:13]:
```suggestion
	<groupId>dev.langchain4j</groupId>
```

File: native-java-gemini-function-calling-example/pom.xml [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+			<version>${langchain4j.version}</version>
+		</dependency>
+		<dependency>
+			<groupId>dev.langchain4j</groupId>
+			<artifactId>langchain4j-vertex-ai-gemini</artifactId>
+			<version>${langchain4j.version}</version>
+		</dependency>
+		<dependency>
+			<groupId>dev.langchain4j</groupId>
+			<artifactId>langchain4j</artifactId>
```
Comment (langchain4j) [inline native-java-gemini-function-calling-example/pom.xml:64]: minor: this dependency is not required, it is imported transitively with `langchain4j-spring-boot-starter`

File: native-java-gemini-function-calling-example/pom.xml [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+			<version>${langchain4j.version}</version>
+		</dependency>
+		<dependency>
+			<groupId>dev.langchain4j</groupId>
+			<artifactId>langchain4j-core</artifactId>
```
Comment (langchain4j) [inline native-java-gemini-function-calling-example/pom.xml:69]: minor: this dependency is not required, it is imported transitively with all other dependencies

File: native-java-gemini-function-calling-example/src/main/java/services/gemini/Langchain4JFunctionCallingApplication.java [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+				new Transaction("003"), new Status("rejected"));
+
+		@Tool("Get the status of a payment transaction")
+		public Status paymentStatus(@P("The id of the payment transaction") String transaction) {
+			System.out.println();
+			return DATASET.get(new Transaction(transaction));
+		}
+	}
+
+	// @AiService
```
Comment (langchain4j) [inline native-java-gemini-function-calling-example/src/main/java/services/gemini/Langchain4JFunctionCallingApplication.java:71]: is it commented out on purpose?

File: native-java-gemini-function-calling-example/README.md [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+
+private static final Map<Transaction, Status> DATASET =
+	Map.of(
+		new Transaction("001"), new Status("pending"),
+		new Transaction("002"), new Status("approved"),
+		new Transaction("003"), new Status("rejected"));
+
+@Bean
+@Description("Get the status of a payment transaction")
+public Function<Transaction, Status> paymentStatus() {
```
Comment (langchain4j) [inline native-java-gemini-function-calling-example/README.md:34]: This does not seem right, should be something like 

```java
@Tool
Status getPaymentStatus(...) {...}
```

File: native-java-gemini-function-calling-example/README.md [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+    <version>${langchain4j.version}</version>
+</dependency>
+<dependency>
+    <groupId>dev.langchain4j</groupId>
+    <artifactId>langchain4j</artifactId>
+    <version>${langchain4j.version}</version>
+</dependency>
+<dependency>
+    <groupId>dev.langchain4j</groupId>
+    <artifactId>langchain4j-core</artifactId>
```
Comment (langchain4j) [inline native-java-gemini-function-calling-example/README.md:63]: minor: this dependency is not required, it is imported transitively with all other dependencies

File: native-java-gemini-function-calling-example/README.md [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+    <version>${langchain4j.version}</version>
+</dependency>
+<dependency>
+    <groupId>dev.langchain4j</groupId>
+    <artifactId>langchain4j-vertex-ai-gemini</artifactId>
+    <version>${langchain4j.version}</version>
+</dependency>
+<dependency>
+    <groupId>dev.langchain4j</groupId>
+    <artifactId>langchain4j</artifactId>
```
Comment (langchain4j) [inline native-java-gemini-function-calling-example/README.md:58]: minor: this dependency is not required, it is imported transitively with `langchain4j-spring-boot-starter`

**Review by langchain4j — COMMENTED** [review]
@ddobrin thanks a lot!

**Comment by ddobrin** [comment]
Thank you for the feedback, code and doc have been updated

**Review by langchain4j — APPROVED** [review]
@ddobrin thank you!

**Pull request merged**